src/tests/parser_test.py

```diff
@@ -562,20 +563,70 @@ def test_matrix_multiplication_with_decorators(code):
     assert inner_function.decorators[0].name == 'a'
 
 
-def test_module_publicity():
-    """Test that a module that has a single leading underscore is private."""
+@pytest.mark.parametrize("parent_path", (
+    Path("package") / "another_package",
+    Path("")
+))
+def test_module_publicity(parent_path):
+    """Test module publicity.
+
+    Module names such as my_module.py are considered public.
+
+    Module names starting with single or double-underscore are private.
+    For example, _my_private_module.py and __my_private_module.py.
+
+    While special "dunder" modules,
+    with leading and trailing double-underscores (e.g. __init__.py) are public.
+
+    The same rules for module name publicity applies to package name publicity.
+
+    Any module within a private package is considered private.
+    """
     parser = Parser()
     code = CodeSnippet("")
 
-    module = parser.parse(code, "filepath")
+    # Public assertions
+    # ===========================================================================
+    module = parser.parse(code, str(parent_path / "module"))
     assert module.is_public
 
-    module = parser.parse(code, "_filepath")
-    assert not module.is_public
+    module = parser.parse(code, str(
+        parent_path / "package" / "__init__"))
+    assert module.is_public
 
-    module = parser.parse(code, "__filepath")
+    module = parser.parse(code, str(
+        parent_path / "__dunder__" / "package" / "module"))
     assert module.is_public
 
+    # Private assertions
+    # ===========================================================================
+
+    # Single Underscore
+    # -----------------
+    module = parser.parse(code, str(parent_path / "_private_module"))
+    assert not module.is_public
+
+    module = parser.parse(code, str(
+        parent_path / "_private_package" / "module"))
+    assert not module.is_public
+
+    module = parser.parse(code, str(
+        parent_path / "_private_package" / "package" / "module"))
+    assert not module.is_public
+
+    # Double Underscore
+    # -----------------
+    module = parser.parse(code, str(parent_path / "__private_module"))
+    assert not module.is_public
+
+    module = parser.parse(code, str(
+        parent_path / "__private_package" / "module"))
+    assert not module.is_public
+
+    module = parser.parse(code, str(
+        parent_path / "__private_package" / "package" / "module"))
```

Comment:
Shouldn't all these file paths end with `.py`? (I know that the code will correctly handle it since we call Path.stem , but just saying since someone might modify it in the future and since we do check for endswith :) )

Reply:
Good point @samj1912.

I can update this. :)

Reply:
Added `.py` file extensions to test path examples in https://github.com/PyCQA/pydocstyle/pull/494/commits/cec57934bf781f3d4956bdcc2c8c62bd6123cc73.